ml-agents/mlagents/trainers/torch/attention.py

```diff
@@ -1,75 +1,55 @@
 from mlagents.torch_utils import torch
 from typing import Tuple, Optional, List
-from mlagents.trainers.torch.layers import LinearEncoder
+from mlagents.trainers.torch.layers import LinearEncoder, Initialization, linear_layer
 
 
 class MultiHeadAttention(torch.nn.Module):
     """
     Multi Head Attention module. We do not use the regular Torch implementation since
     Barracuda does not support some operators it uses.
     Takes as input to the forward method 3 tensors:
-     - query: of dimensions (batch_size, number_of_queries, key_size)
-     - key: of dimensions (batch_size, number_of_keys, key_size)
-     - value: of dimensions (batch_size, number_of_keys, value_size)
+     - query: of dimensions (batch_size, number_of_queries, embedding_size)
+     - key: of dimensions (batch_size, number_of_keys, embedding_size)
+     - value: of dimensions (batch_size, number_of_keys, embedding_size)
     The forward method will return 2 tensors:
-     - The output: (batch_size, number_of_queries, output_size)
+     - The output: (batch_size, number_of_queries, embedding_size)
      - The attention matrix: (batch_size, num_heads, number_of_queries, number_of_keys)
     """
 
     NEG_INF = -1e6
 
-    def __init__(
-        self,
-        query_size: int,
-        key_size: int,
-        value_size: int,
-        output_size: int,
-        num_heads: int,
-        embedding_size: int,
-    ):
+    def __init__(self, embedding_size: int, num_heads: int):
         super().__init__()
         self.n_heads, self.embedding_size = num_heads, embedding_size
```

Comment:
I think `self.embedding_size` should be equal to `self.embedding_size // self.n_heads * self.n_heads` to avoid rounding issues in division